### ISSUE TITLE
References to issues should be re-written when synced

### ISSUE THREAD

**Issue by ericallam** [calcom/synclinear.com#113]
For example, here is an issue I just created in linear that was synced to GitHub:

<img width="1079" alt="CleanShot 2023-07-13 at 20 02 50@2x" src="https://github.com/calcom/synclinear.com/assets/534/f8b1f8f5-d661-4ca6-a6f3-d0fed0efebf1">

That `[TR-654]` links back to Linear, instead of linking to the synced issue in GitHub. It would be great if those reference links could automatically be rewritten to reference the correct GitHub issue (or if the linear issue references isn't public, then it should just keep the link as is now)

**Comment by tedspare** [comment]
Interesting! One possible approach:
- Include a RegEx check-and-replace for Linear ticket refs ("ABC-123" where "ABC" is known at runtime as `linearTeamId`) in `prepareMarkdownContent()`
- Check the DB for a GH issue corresponding to that ticket via the `SyncedIssues` table where `linearIssueId` is known
- For each Regex match, add a GH link if possible
- Watch out for infinite loops if the content is different on each side

**Comment by ericallam** [comment]
Is checking that the content is equal the way you prevent infinite loops?

**Comment by tedspare** [comment]
That's right, at least after SyncLinear's transformations: `linearMarkdown === syncLinear(githubMarkdown)`
